python/lsst/daf/base/baseLib.py

```diff
@@ -85,18 +135,23 @@ def _propertyContainerAdd(container, name, value, typeMenu, *args):
         exemplar = value
 
     t = type(exemplar)
-    if t in typeMenu:
-        return getattr(container, "add" + typeMenu[t])(name, value, *args)
+    addType = _guessIntegerType(container, name, exemplar)
+
+    if addType is not None or t in typeMenu:
+        if addType is None:
+            addType = typeMenu[t]
+        return getattr(container, "add" + addType)(name, value, *args)
     # Allow for subclasses
     for checkType in typeMenu:
         if isinstance(exemplar, checkType):
             return getattr(container, "add" + typeMenu[checkType])(name, value, *args)
     raise lsst.pex.exceptions.TypeError("Unknown value type for %s: %s" % (name, t))
 
+
 # Mapping of type to method names
+from past.builtins import long
 _PS_typeMenu = {bool: "Bool",
                 int: "Int",
-                long: "LongLong",
```

Comment:
If `long` is being removed from the list completely how will this work on Python 2? (and why have the `long` import if it's been removed).